wire-ios-sync-engine/Source/Data Model/Conversation+AccessMode.swift

```diff
@@ -77,11 +82,17 @@ extension ZMConversation {
         let request = WirelessRequestFactory.fetchLinkRequest(for: self, apiVersion: apiVersion)
         request.add(ZMCompletionHandler(on: managedObjectContext!) { response in
             if response.httpStatus == 200,
-               let uri = response.payload?.asDictionary()?[ZMConversation.TransportKey.uri] as? String {
-                completion(.success(uri))
+               let payloadDict = response.payload?.asDictionary() {
+                let hasPassword = payloadDict["has_password"] as? Bool ?? false
+                if let uri = payloadDict[ZMConversation.TransportKey.uri] as? String {
+                    completion(.success((uri: uri, secured: hasPassword)))
+                } else {
+                    completion(.failure(WirelessLinkError.invalidResponse))
+                }
             } else if response.httpStatus == 404 {
```

Comment:
💅 Indentations looks off

Reply:
I tried to fix it through Xcode but as it turns out it is as it supposed to be.